azure/application-gateway/advanced/variables.tf

```diff
@@ -369,4 +369,119 @@ variable "waf_configuration" {
     })), {})
   })
   default = null
+}
+
+variable "listener_waf_configuration" {
+  description = "Defining the WAF policy per listener"
+  type = map(object({ // Key = policy name
+    associated_listeners = list(string)
+    firewall_mode        = optional(string, "Prevention")
+
+    enable_OWASP           = optional(bool, true)
+    OWASP_rule_set_version = optional(string, "3.2")
+    OWASP_rule_group_override = optional(map(map(object({
+      enabled = optional(bool, true)
+      action  = optional(string)
+    }))), {})
+
+    enable_Microsoft_BotManagerRuleSet           = optional(bool, false)
+    Microsoft_BotManagerRuleSet_rule_set_version = optional(string, "1.0")
+    Microsoft_BotManagerRuleSet_rule_group_override = optional(map(map(object({
+      enabled = optional(bool, true)
+      action  = optional(string)
+    }))), {})
+
+    file_upload_limit_mb     = optional(number, 500)
+    max_request_body_size_kb = optional(number, 128)
+    managed_rule_exclusion = optional(list(object({
+      match_variable          = string
+      selector_match_operator = optional(string)
+      selector                = optional(string)
+    })), [])
+    custom_rules = optional(map(object({
+      priority = number
+      action   = optional(string, "Block")
```

Comment:
should this be Log by default?

Reply:
Most LLD's that I've come across have (when a custom rule is specified) specified that Block is the action. 

If you believe this should be more explicit, then perhaps we remove the "optional" variable entirely and make it mandatory?